tests/integration/vpp/040_acl_test.go

```diff
@@ -0,0 +1,614 @@
+			Ip: &acl.ACL_Rule_IpRule_Ip{
+				SourceNetwork:      "",
+				DestinationNetwork: "",
+			},
+		},
+	},
+	{
+		//RuleName:  "denyICMP",
+		Action: acl.ACL_Rule_DENY,
+		IpRule: &acl.ACL_Rule_IpRule{
+			Icmp: &acl.ACL_Rule_IpRule_Icmp{
+				Icmpv6: false,
+				IcmpCodeRange: &acl.ACL_Rule_IpRule_Icmp_Range{
+					First: 150,
+					Last:  250,
+				},
+				IcmpTypeRange: &acl.ACL_Rule_IpRule_Icmp_Range{
+					First: 1150,
+					Last:  1250,
+				},
+			},
+		},
+	},
+	{
+		//RuleName:  "denyICMPv6",
+		Action: acl.ACL_Rule_DENY,
+		IpRule: &acl.ACL_Rule_IpRule{
+			Icmp: &acl.ACL_Rule_IpRule_Icmp{
+				Icmpv6: true,
+				IcmpCodeRange: &acl.ACL_Rule_IpRule_Icmp_Range{
+					First: 150,
+					Last:  250,
+				},
+				IcmpTypeRange: &acl.ACL_Rule_IpRule_Icmp_Range{
+					First: 1150,
+					Last:  1250,
+				},
+			},
+		},
+	},
+	{
+		//RuleName:  "permitTCP",
+		Action: acl.ACL_Rule_PERMIT,
+		IpRule: &acl.ACL_Rule_IpRule{
+			Tcp: &acl.ACL_Rule_IpRule_Tcp{
+				TcpFlagsMask:  20,
+				TcpFlagsValue: 10,
+				SourcePortRange: &acl.ACL_Rule_IpRule_PortRange{
+					LowerPort: 150,
+					UpperPort: 250,
+				},
+				DestinationPortRange: &acl.ACL_Rule_IpRule_PortRange{
+					LowerPort: 1150,
+					UpperPort: 1250,
+				},
+			},
+		},
+	},
+	{
+		//RuleName:  "denyUDP",
+		Action: acl.ACL_Rule_DENY,
+		IpRule: &acl.ACL_Rule_IpRule{
+			Udp: &acl.ACL_Rule_IpRule_Udp{
+				SourcePortRange: &acl.ACL_Rule_IpRule_PortRange{
+					LowerPort: 150,
+					UpperPort: 250,
+				},
+				DestinationPortRange: &acl.ACL_Rule_IpRule_PortRange{
+					LowerPort: 1150,
+					UpperPort: 1250,
+				},
+			},
+		},
+	},
+}
+
+var aclMACIPrules = []*acl.ACL_Rule{
+	{
+		//RuleName:  "denyIPv4",
+		Action: acl.ACL_Rule_DENY,
+		MacipRule: &acl.ACL_Rule_MacIpRule{
+			SourceAddress:        "192.168.0.1",
+			SourceAddressPrefix:  uint32(16),
+			SourceMacAddress:     "11:44:0A:B8:4A:35",
+			SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+		},
+	},
+	{
+		//RuleName:  "denyIPv6",
+		Action: acl.ACL_Rule_DENY,
+		MacipRule: &acl.ACL_Rule_MacIpRule{
+			SourceAddress:        "dead::1",
+			SourceAddressPrefix:  uint32(64),
+			SourceMacAddress:     "11:44:0A:B8:4A:35",
+			SourceMacAddressMask: "ff:ff:ff:ff:00:00",
+		},
+	},
+}
+
+func TestCRUDIPAcl(t *testing.T) {
+	ctx := setupVPP(t)
+	defer ctx.teardownVPP()
+
+	ih := ifplugin_vppcalls.CompatibleInterfaceVppHandler(ctx.vppBinapi, logrus.NewLogger("test"))
+
+	const ifName = "loop1"
+	ifIdx, errI := ih.AddLoopbackInterface(ifName)
+	if errI != nil {
+		t.Fatalf("creating interface failed: %v", errI)
+	}
+	t.Logf("interface created %v", ifIdx)
+
+	ifIndexes := ifaceidx.NewIfaceIndex(logrus.NewLogger("test-iface1"), "test-iface1")
+	ifIndexes.Put(ifName, &ifaceidx.IfaceMetadata{
+		SwIfIndex: ifIdx,
+	})
+
+	h := aclplugin_vppcalls.CompatibleACLVppHandler(ctx.vppBinapi, ifIndexes, logrus.NewLogger("test"))
+	if h == nil {
+		t.Fatalf("handler was not created")
+	}
+
+	acls, errx := h.DumpACL()
+	if errx != nil {
```

Comment:
Let's be consistent and use `Expect(...)...` for all of these from this point below.